src/components/subsetDraft/SubsetMetadata.jsx

```diff
@@ -1,95 +1,152 @@
-                           htmlFor="from_date">{t('Valid from')}: </label>
-                    <DatePicker id='from_date' style={{display: 'block'}}
-                                value={subset.draft.validFrom}
-                                onChange={(date) => subset.dispatch({action: 'from', data: date})}
-                                clearIcon={null}
-                                format='dd.MM.y'
-                                locale={languages.find(i => i.default).IETF}
-                                className='datepicker'/>
+            <section style={{ margin: "5px 0 5px 0" }}>
+                <div
+                    style={{ float: "left", marginRight: "20px", padding: "0" }}
+                >
+                    <label
+                        style={{
+                            display: "block",
+                            fontSize: "16px",
+                            fontFamily: "Roboto"
+                        }}
+                        htmlFor='from_date'
+                    >
+                        {t("Valid from")}:{" "}
+                    </label>
+                    <DatePicker
+                        id='from_date'
+                        style={{ display: "block" }}
+                        value={subset.draft.validFrom}
+                        onChange={date => handleDateChange(date, 'from')}
+                        clearIcon={null}
+                        format='dd.MM.y'
+                        locale={languages.find(i => i.default).IETF}
+                        className='datepicker'
+                    />
                 </div>
 
-                <div style={{float: 'left'}}>
-                    <label style={{display: 'block', fontSize: '16px', fontFamily: 'Roboto'}}
-                           htmlFor="to_date">{t('Valid to')}: </label>
-                    <DatePicker id='to_date' style={{display: 'block'}}
-                                value={subset.draft.validUntil}
-                                onChange={(date) => subset.dispatch({action: 'to', data: date})}
-                                clearIcon={null}
-                                format='dd.MM.y'
-                                locale={languages.find(i => i.default).IETF}
-                                className='datepicker'/>
+                <div style={{ float: "left" }}>
+                    <label
+                        style={{
+                            display: "block",
+                            fontSize: "16px",
+                            fontFamily: "Roboto"
+                        }}
+                        htmlFor='to_date'
+                    >
+                        {t("Valid to")}:{" "}
+                    </label>
+                    <DatePicker
+                        id='to_date'
+                        style={{ display: "block" }}
+                        value={subset.draft.validUntil}
+                        onChange={date => handleDateChange(date,'to')}
```

Comment:
Here the component does not know if the AppContext want to show Modal or can throw validation error. 
The only thing it knows, is the date the user has chosen. It informs the AppContext about this action by calling `subset.dispatch({action "to", date})`.